src/query_router.rs

```diff
@@ -11,6 +12,7 @@ use sqlparser::ast::{
 };
 use sqlparser::dialect::PostgreSqlDialect;
 use sqlparser::parser::Parser;
+use std::sync::OnceLock;
```

Comment:
We probably want to use the async version of this construct to avoid blocking the worker threads.

Reply:
A few reasons I'm wondering if it's worth it:
1. This'll only happen once since it's initializing a static object.
2. The init itself happens all in memory.
3. The init uses all sync methods so i'll have to wrap it with an async method that doesn't have anything to await on.

wdyt?

Reply:
@drdrsh what do you think?